flang-rt/lib/runtime/derived.cpp

```diff
@@ -106,16 +88,41 @@ RT_API_ATTRS int InitializeTicket::Continue(WorkQueue &workQueue) {
       GetComponentExtents(extents, *component_, instance_);
       Descriptor &compDesc{componentDescriptor_.descriptor()};
       const typeInfo::DerivedType &compType{*component_->derivedType()};
-      compDesc.Establish(compType,
-          instance_.ElementComponent<char>(subscripts_, component_->offset()),
-          component_->rank(), extents);
-      Advance();
+      compDesc.Establish(compType, rawComponent, component_->rank(), extents);
       if (int status{workQueue.BeginInitialize(compDesc, compType)};
           status != StatOk) {
+        SkipToNextComponent();
         return status;
       }
-    } else {
-      SkipToNextComponent();
+    }
+  }
+  // The first element is now complete.  Copy it into the others.
+  if (elements_ < 2) {
+  } else {
+    auto elementBytes{static_cast<SubscriptValue>(instance_.ElementBytes())};
+    if (auto stride{instance_.FixedStride()}) {
+      if (*stride == elementBytes) { // contiguous
+        for (std::size_t done{1}; done < elements_;) {
+          std::size_t chunk{elements_ - done};
+          if (chunk > done) {
+            chunk = done;
+          }
+          char *uninitialized{rawInstance + done * *stride};
+          std::memcpy(uninitialized, rawInstance, chunk * *stride);
```

Comment:
As the chunk size grows, I guess, we can hit cache conflicts depending on the aliasing implementation.  Do you know if there is any non-temporal memcpy implementation that we can try?

Reply:
Not sure about that.  It will probably be the same as memcpy if the compiler vectorizes it.  I think memcpy may do better with `rep movs` on x86.  Let's keep it simple with memcpy.  If the cache issue pops up anywhere I will experiment with https://clang.llvm.org/docs/LanguageExtensions.html#non-temporal-load-store-builtins

Reply:
`memcpy` is not defined for overlapping regions, unfortunately.